src/graphics/Screen.cpp

```diff
@@ -2133,9 +2240,12 @@ void Screen::setFrames()
     if (devicestate.has_rx_text_message && shouldDrawMessage(&devicestate.rx_text_message)) {
         normalFrames[numframes++] = drawTextMessageFrame;
     }
-    // If we have a waypoint - show it next, unless it's a phone message and we aren't using any special modules
+
+    // If we have a waypoint (not expired, not deleted)
     if (devicestate.has_rx_waypoint && shouldDrawMessage(&devicestate.rx_waypoint)) {
```

Comment:
IMO, drop the shouldDrawMessage() check here. Navigating to a self-dropped waypoint seems like a good thing.

Reply:
Makes sense to me!